include/itkImageToTorchImageFilter.hxx

```diff
@@ -53,23 +52,23 @@ ImageToTorchImageFilter< TInputImage, TOutputTorchImage >
 }
 
 
-template< typename TInputImage, typename TOutputTorchImage >
+template< typename TInputImage >
 void
-ImageToTorchImageFilter< TInputImage, TOutputTorchImage >
-::BeforeThreadedGenerateData()
+ImageToTorchImageFilter< TInputImage >
+::GenerateData()
 {
   // Call the superclass' implementation of this method
-  Superclass::BeforeThreadedGenerateData();
+  Superclass::GenerateData();
 
   // Get input and output and check that their sizes match
   typename Self::InputImageType const * const inputImage = this->GetInput();
-  const typename Self::InputImageSizeType inputSize = inputImage->GetBufferedRegion().GetSize();
+  const typename Self::InputImageSizeType inputBufferedRegionSize = inputImage->GetBufferedRegion().GetSize();
   typename Self::InputImagePixelType const * const inputBufferPointer = inputImage->GetBufferPointer();
 
   typename Self::OutputTorchImageType * const outputTorchImage = this->GetOutput();
-  const typename Self::OutputTorchImageSizeType outputSize = outputTorchImage->GetBufferedRegion().GetSize();
+  const typename Self::OutputTorchImageSizeType outputBufferedRegionSize = outputTorchImage->GetBufferedRegion().GetSize();
 
-  itkAssertOrThrowMacro( inputSize == outputSize, "ImageToTorchImageFilter: input and output images must have the same size in each dimension");
+  itkAssertOrThrowMacro( inputBufferedRegionSize == outputBufferedRegionSize, "ImageToTorchImageFilter: input and output images must have the same size in each dimension");
```

Comment:
I meant the error message should mentioned buffered region, e.g.: _ImageToTorchImageFilter: input and output images' buffered regions must have the same size in each dimension_

Reply:
Will do.